rct229/rules/section5/section5rule49.py

```diff
@@ -0,0 +1,145 @@
+
+            building_total_envelope_area = sum(
+                [
+                    surface.get("area", ZERO.AREA)
+                    for surface in find_all("$..surfaces[*]", building_p)
+                    if scc_dict_p[surface["id"]] != SCC.UNREGULATED
+                ],
+                ZERO.AREA,
+            )
+
+            for zone in find_all("$..zones[*]", building_p):
+                if zcc_dict_p[zone["id"]] in [
+                    ZCC.CONDITIONED_RESIDENTIAL,
+                    ZCC.CONDITIONED_NON_RESIDENTIAL,
+                    ZCC.CONDITIONED_MIXED,
+                    ZCC.SEMI_HEATED,
+                ]:
+                    building_total_air_leakage_rate += zone["infiltration"][
+                        "infiltration_flow_rate"
+                    ]
+
+                measured_air_leakage_rate = zone["infiltration"].get(
+                    "measured_air_leakage_rate"
+                )
+                if measured_air_leakage_rate:
+                    building_total_measured_air_leakage_rate += zone["infiltration"][
+                        "measured_air_leakage_rate"
+                    ]
+                else:
+                    empty_measured_air_leakage_rate_flow_flag = True
+
+            target_air_leakage_rate_75pa_p = (
+                TARGET_AIR_LEAKAGE_COEFF
+            ) * building_total_envelope_area
+
+            return {
+                "building_total_air_leakage_rate": building_total_air_leakage_rate,
+                "building_total_measured_air_leakage_rate": building_total_measured_air_leakage_rate,
+                "target_air_leakage_rate_75pa_p": target_air_leakage_rate_75pa_p,
+                "empty_measured_air_leakage_rate_flow_flag": empty_measured_air_leakage_rate_flow_flag,
+            }
+
+        def rule_check(self, context, calc_vals, data=None):
+            building_total_air_leakage_rate = calc_vals[
+                "building_total_air_leakage_rate"
+            ]
+            building_total_measured_air_leakage_rate = calc_vals[
+                "building_total_measured_air_leakage_rate"
+            ]
+            target_air_leakage_rate_75pa_p = calc_vals["target_air_leakage_rate_75pa_p"]
+            empty_measured_air_leakage_rate_flow_flag = calc_vals[
+                "empty_measured_air_leakage_rate_flow_flag"
+            ]
+
+            return std_equal(
+                building_total_air_leakage_rate,
+                0.112 * target_air_leakage_rate_75pa_p,
+            ) or (
+                building_total_air_leakage_rate
+                != 0.112 * target_air_leakage_rate_75pa_p
+                and empty_measured_air_leakage_rate_flow_flag == False
+                and std_equal(
+                    building_total_air_leakage_rate,
+                    0.112 * building_total_measured_air_leakage_rate,
+                )
+            )
+
+        def manual_check_required(self, context, calc_vals=None, data=None):
```

Comment:
Since `manual_check_required()` will be executed before `rule_check()`, it would be nicer to place it above `rule_check()` in this file. That makes it easier to keep the correct logic flow in mind.